network/manager.go

```diff
@@ -436,24 +431,28 @@ func (nm *networkManager) UpdateEndpointState(ep *endpoint) error {
 
 // GetEndpointState will make a call to CNS GetEndpointState API in the stateless CNI mode to fetch the endpointInfo
 // TODO unit tests need to be added, WorkItem: 26606939
-func (nm *networkManager) GetEndpointState(networkID, endpointID string) (*EndpointInfo, error) {
-	endpointResponse, err := nm.CnsClient.GetEndpoint(context.TODO(), endpointID)
+// In stateless cni, container id is the endpoint id, so you can pass in either
+func (nm *networkManager) GetEndpointState(networkID, containerID string) ([]*EndpointInfo, error) {
+	endpointResponse, err := nm.CnsClient.GetEndpoint(context.TODO(), containerID)
 	if err != nil {
-		return nil, errors.Wrapf(err, "Get endpoint API returend with error")
+		return nil, errors.Wrapf(err, "Get endpoint API returned with error")
 	}
-	epInfo := cnsEndpointInfotoCNIEpInfo(endpointResponse.EndpointInfo, endpointID)
-	if epInfo.IsEndpointStateIncomplete() {
-		if networkID == "" {
-			networkID = DefaultNetworkID
-		}
-		epInfo, err = epInfo.GetEndpointInfoByIPImpl(epInfo.IPAddresses, networkID)
-		if err != nil {
-			return nil, errors.Wrapf(err, "Get endpoint API returend with error")
+	epInfos := cnsEndpointInfotoCNIEpInfos(endpointResponse.EndpointInfo, containerID)
+
+	for i := 0; i < len(epInfos); i++ {
+		if epInfos[i].NICType == cns.InfraNIC {
```

Comment:
The NICType needs to be filled out manually in case we have incomplete state from Statefull CNI.

Reply:
This should be addressed in `cnsEndpointInfotoCNIEpInfo` if the`ifName == ""`, we populate the nictype with infra nic